x/emissions/module/abci.go

```diff
@@ -82,15 +82,19 @@ func EndBlocker(ctx context.Context, am AppModule) error {
 					MaxUnfulfilledReputerRequests = types.DefaultParams().MaxUnfulfilledReputerRequests
 					fmt.Println("Error getting max retries to fulfil nonces for worker requests (using default), err:", err)
 				}
-				reputerPruningBlock := blockHeight - int64(MaxUnfulfilledReputerRequests)*topic.EpochLength
-				fmt.Println("Pruning reputer nonces before block: ", reputerPruningBlock, " for topic: ", topic.Id, " on block: ", blockHeight)
-				am.keeper.PruneReputerNonces(sdkCtx, topic.Id, reputerPruningBlock)
+				reputerPruningBlock := blockHeight - (int64(MaxUnfulfilledReputerRequests)*topic.EpochLength + topic.GroundTruthLag)
```

Comment:
related to this code but not exactly this code: I believe you also need to do a similar update to `pruneRecordsAfterRewards()` in `rewards.go` as well

Reply:
Checked the code, I think that can stay as is. 

The purpose of the pruneRecordsAfterRewards is to prune before any unfulfilledReputerNonce. 

However, the reputer nonces with gt lag remain unfulfilled for a longer time. So regardless that time, they will remain in there and oldest one will mark the limit in the same way = still works.